### PR TITLE
Regain control over JavaScript fatigue

### PR DESCRIPTION
**Pull request by hawkaa**
Okay, so this Pull Request manifests that I have finally been able to overcome the Javascript fatigue.

In summary:
- I've regained control over hot reloading
- I've created a production web server that does not use hot reloading
- I'm building the index.html using a template

Please give thumbs up!!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,7 +9,8 @@
   "scripts": {
     "build": "webpack -p --config webpack.production.config.js",
     "start": "babel-node server/server.js",
-    "lint": "eslint --ignore-path .gitignore .",
+    "start-production": "NODE_ENV=production node dist/server.js",
```
Comment (johnolos) [inline package.json:12]: Consider adding [foreverjs](https://github.com/foreverjs/forever) as a dependency and running our production-server with it.

File: server/server.js [hunk 2]
```diff
@@ -1,26 +1,30 @@
...
 const app = express();
-const indexPath = path.join(__dirname, '../public/index.html');
-const publicPath = express.static(path.join(__dirname, '../dist/assets'));
 
-app.use('/assets', publicPath);
-app.get('/', (_, res) => { res.sendFile(indexPath); });
 
-// setup debugging
 if (process.env.NODE_ENV !== 'production') {
+  console.log('Running in development mode with webpack hot realoding middleware.');
```
Comment (johnolos) [inline server/server.js]: I thought eslint prohibited console.logs?
Anyway, consider using winston when logging.

Comment (johnolos) [inline server/server.js]: typo: 'reloading'

File: src/App.jsx [hunk 3]
```diff
@@ -1,8 +1,4 @@
-import React, {Component} from 'react';
+import React from 'react';
 
-class App extends Component {
-  render() {
-    return (<h1>Welcome to Pubcrawler</h1>);
-  }
-}
+const App = () => <h1>Welcome to pubcrawler</h1>;
```
Comment (johnolos) [inline src/App.jsx:3]: Perfectly valid syntax.
It's a matter of preference, but `export default class App {...};` could be used in this instance given this `.js`-file will never contain anything other than entry point to our react-app( and perhaps routing strategy).

File: src/index.jsx [hunk 4]
```diff
@@ -2,24 +2,23 @@ import { AppContainer } from 'react-hot-loader';
...
-  rootEl
+  rootEl,
 );
 
 if (module.hot) {
   module.hot.accept('./App', () => {
-    // If you use Webpack 2 in ES modules mode, you can
-    // use <App /> here rather than require() a <NextApp />.
-    const NextApp = require('./App').default;
+    const NextApp = require('./App').default; // eslint-disable-line global-require
```
Comment (johnolos) [inline src/index.jsx:16]: Is it possible to rewrite this code-block without having to disable eslint?

Comment (johnolos) [reply src/index.jsx:16]: I don't think you have to require(...).**_default_** if you use the syntax mentioned above.

File: webpack.production.config.js [hunk 5]
```diff
@@ -15,11 +17,37 @@ module.exports = {
...
         loader: 'babel-loader',
         options: { presets: ['latest', 'react'] },
       }],
     }],
   },
-};
+  plugins: [
+    new HtmlWebpackPlugin({
+      template: path.resolve(__dirname, 'src', 'index.ejs'),
+    }),
```
Comment (johnolos) [inline webpack.production.config.js:30]: Good job on setting up webpack. :+1: 
I know it can be a struggle in itself.

**Review by johnolos — APPROVED** [review]
Pretty good!

Maybe a little nitpicky of me with some of my comments.

:rocket: when your happy.

File: package.json (hunk 1 above)
Comment (hawkaa) [reply package.json:12]: Lets hold the `foreverjs` until we gain some experience with Heroku and the dynos. Ideally, we should use static site hosting, and not an express backend.

File: server/server.js (hunk 2 above)
Comment (hawkaa) [reply server/server.js]: Just warnings. Will add winston in a separate PR.

File: src/App.jsx (hunk 3 above)
Comment (hawkaa) [reply src/App.jsx:3]: Well, the linter complained, so I turned it into a stateless function. I expect this to change pretty soon.

File: webpack.production.config.js (hunk 5 above)
Comment (hawkaa) [reply webpack.production.config.js:30]: Thanks 👍

File: src/index.jsx (hunk 4 above)
Comment (hawkaa) [reply src/index.jsx:16]: I tried to rewrite, but I couldn't get it to work... I'll see if I can remove the `default`.

Comment (hawkaa) [reply src/index.jsx:16]: Seems like I need `default`.

**Pull request merged**